app/src/main/java/com/tokbox/android/demo/learningopentok/ChatActivity.java

```diff
@@ -70,13 +72,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
     }
 
     private void initializeSession() {
-        mSession = new Session(this, mApiKey, mSessionId);
+        mSession = (new Session.Builder(this, mApiKey, mSessionId).build());
         mSession.setSessionListener(this);
         mSession.connect(mToken);
     }
 
     private void initializePublisher() {
-        mPublisher = new Publisher(this);
+        mPublisher = (new Publisher.Builder(this)).build();
```

Comment:
I think we should remove a pair of parens:

``` mPublisher = new Publisher.Builder(this).build();```

(Note that you group the additional parens differently here than you do in the `Session.Builder` line.)